plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -786,6 +798,9 @@ public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHa
     @Override
     public Optional<ParameterizedExpression> convertPredicate(ConnectorSession session, ConnectorExpression expression, Map<String, ColumnHandle> assignments)
     {
+        if (isEnableStringPushdownWithCollate(session)) {
+            return connectorExpressionRewriterWithCollate.rewrite(session, expression, assignments);
```

Comment:
nice

Reply:
thanks. i am not super proud of this part, but i didn't find a better way to do this

Reply:
-> https://github.com/trinodb/trino/pull/20318